app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -162,7 +169,16 @@ class PatientSummaryScreen(context: Context, attrs: AttributeSet) : RelativeLayo
   }
 
   private fun setupEditButtonClicks() {
-    editButton.setOnClickListener { screenRouter.push(createEditPatientScreenKey(patientSummaryProfile!!)) }
+    editButton.setOnClickListener {
+      val (key, architecture) = if (mobiusMigrationConfig.useEditPatientScreen) {
+        createEditPatientScreenKey(patientSummaryProfile!!) to Architecture.MOBIUS
+      } else {
+        createPatientEditScreenKey(patientSummaryProfile!!) to Architecture.ORIGINAL
+      }
+      Analytics.reportArchitectureMigration("EditPatientScreen", architecture, "PatientSummaryScreen")
```

Comment:
Won't the `hostScreen` parameter be `null` since the `EditPatientScreen` is a discrete screen and not a part of some other screen?

Reply:
Good catch, done.